crates/viewer/re_view_spatial/src/max_image_dimension_subscriber.rs

```diff
@@ -72,96 +81,129 @@ impl PerStoreChunkSubscriber for MaxImageDimensionsStoreSubscriber {
                 continue;
             }
 
-            // Handle `Image`, `DepthImage`, `SegmentationImage`…
             let chunk = &event.diff.chunk;
             let components = chunk.components();
-            for image_format_list_array in components.get_by_component_name(ImageFormat::name()) {
-                for new_dim in image_format_list_array.iter().filter_map(|array| {
-                    array.and_then(|array| {
-                        let array = arrow::array::ArrayRef::from(array);
-                        ImageFormat::from_arrow(&array).ok()?.into_iter().next()
-                    })
-                }) {
-                    let max_dim = self
-                        .max_dimensions
-                        .entry(chunk.entity_path().clone())
-                        .or_default();
-
-                    max_dim.width = max_dim.width.max(new_dim.width);
-                    max_dim.height = max_dim.height.max(new_dim.height);
-                }
-            }
+            let entity_path = chunk.entity_path();
 
-            if components.has_component_with_archetype_name(archetypes::Image::name()) {
-                self.max_dimensions
-                    .entry(chunk.entity_path().clone())
-                    .or_default()
-                    .image_types
-                    .insert(ImageTypes::IMAGE);
+            // Handle new video codecs first since we do a lookup on this later.
+            if components.contains_key(&archetypes::VideoStream::descriptor_codec()) {
+                for codec in chunk.iter_component::<components::VideoCodec>(
+                    &archetypes::VideoStream::descriptor_codec(),
+                ) {
+                    let Some(codec) = codec.first() else {
+                        continue;
+                    }; // Ignore both empty arrays and multiple codecs per row.
+                    if let Some(existing_codec) =
+                        self.video_codecs.insert(entity_path.hash(), *codec)
+                    {
+                        if existing_codec != *codec {
+                            re_log::warn!(
+                                "Overwriting existing video codec for entity path {:?} from {:?} to {:?}. Video codec per entity is expected to never change.",
```

Comment:
```suggestion
                                "Changing video codec for entity path {:?} from {:?} to {:?}. This is unexpected, video codecs should remain constant per entity.",
```
This might make it more clear that this is user error?